index.bs

```diff
@@ -364,11 +411,74 @@ When asked to <dfn export>report the layout shift</dfn> for an active
     1. Set |newEntry|'s <dfn attribute for=LayoutShift>hadRecentInput</dfn>
         attribute to <code>true</code> if {{LayoutShift/lastInputTime}} is less
         than 500 milliseconds in the past, and <code>false</code> otherwise.
+    1. Set |newEntry|'s <dfn attribute for=LayoutShift>sources</dfn> attribute
+        to the result of invoking the algorithm to <a>report the layout shift
+        sources</a> for |D|.
     1. <a href="https://w3c.github.io/performance-timeline/#dfn-queue-a-performanceentry">Queue the PerformanceEntry</a>
         |newEntry| object.
 
 </div>
 
+Report the layout shift sources {#sec-report-layout-shift-sources}
+------------------------------------------------------------------
+
+<div algorithm="report the layout shift sources">
+When asked to <dfn>report the layout shift sources</dfn> for an active
+{{Document}} |D|, run the following steps:
+
+1. Let |C| be an empty sequence of {{LayoutShiftAttribution}} objects.
+
+1. For each member |N| of the <a>unstable node set</a> of |D|, run these steps:
+
+    1. Let |NA| be the result of running the algorithm to
+        <a>create the attribution</a> for |N| and |D|.
+    1. If there exists any member |existing| of |C| such that the
+        <a>node impact region</a> of |N| is a subset of the
+        <a>node impact region</a> of |existing|, then do nothing.
+    1. Otherwise, if there exists any member |existing| of |C| such
+        that the <a>node impact region</a> of |existing| is a subset
+        of the <a>node impact region</a> of |N|, then replace
+        |existing| with |NA| in |C|.
+    1. Otherwise, if there are fewer than 5 members of |C|, then
+        append |NA| to |C|.
+    1. Otherwise, run these steps:
+
+        1. Let |smallest| be the first member of |C| whose
+            <a>node impact region</a> is not greater in area than
+            the <a>node impact region</a> of any other member of |C|.
+        1. If the area of the <a>node impact region</a> of |N| is
+            greater than the area of the <a>node impact region</a>
+            of |smallest|, then replace |smallest| with |NA| in |C|.
+
+1. Return |C|.
+
+</div>
+
+<div algorithm="create the attribution">
+When asked to <dfn>create the attribution</dfn> for a {{Node}} |N|
+and an active {{Document}} |D|, run the following steps:
+
+1. Create a new {{LayoutShiftAttribution}} object |A|.
+1. Set the <dfn attribute for=LayoutShiftAttribution>node</dfn> attribute
```

Comment:
This seems incorrect. Instead, the attribute getter of `node` needs to call the method (i.e. it needs to be called every time the attribute getter is called). Here, we're setting the value forever to whatever it returns at this point in time, which would still result in a leak.

Reply:
Good catch, fixed.